src/identity.js

```diff
@@ -0,0 +1,56 @@
+
+const isFunction = require('./utils/is-function')
+
+class Identity {
+  constructor (id, publicKey, provider) {
```

Comment:
As per the comment above:

*In order to make it possible to verify that the signing key belongs to the identity that the entry claims to have, the entry key needs to contain a) the original ID of the user (identity.id) b) the public signing key (identity.publicKey) and c) a signature of <id + publicKey> signed by the original ID (eg. an eth address).*

...we need to add a `signature` member variable to Identity. I feel it should be the third argument in the constructor and we should add a getter below similar to other member variables.